crates/sui-indexer/src/models/tx_indices.rs

```diff
@@ -101,37 +114,68 @@ impl TxIndex {
                 cp_sequence_number,
                 tx_sequence_number,
                 recipient: s.to_vec(),
+                transaction_kind,
             })
             .collect();
-        let tx_input_objects = self
-            .input_objects
-            .iter()
-            .map(|o| StoredTxInputObject {
-                cp_sequence_number,
-                tx_sequence_number,
-                object_id: bcs::to_bytes(&o).unwrap(),
-            })
-            .collect();
-        let tx_changed_objects = self
-            .changed_objects
-            .iter()
-            .map(|o| StoredTxChangedObject {
-                cp_sequence_number,
-                tx_sequence_number,
-                object_id: bcs::to_bytes(&o).unwrap(),
-            })
-            .collect();
-        let tx_calls = self
-            .move_calls
-            .iter()
-            .map(|(p, m, f)| StoredTxCalls {
-                cp_sequence_number,
-                tx_sequence_number,
-                package: p.to_vec(),
-                module: m.to_string(),
-                func: f.to_string(),
-            })
-            .collect();
+
+        let mut address_rel_map: HashMap<&SuiAddress, i16> = HashMap::new();
+        for sender in &self.senders {
+            address_rel_map.entry(sender).or_insert(0);
+        }
+
+        for recipient in &self.recipients {
+            address_rel_map
+                .entry(recipient)
+                .and_modify(|e| {
+                    if *e == 0 {
+                        *e = 1
+                    }
+                })
+                .or_insert(2);
+        }
+
+        let mut tx_input_objects = Vec::new();
+
+        for tx_input_object in self.input_objects {
```

Comment:
a bit repetitive, but i feel like writing a trait or generic function is also overkill?